src/types/button.ts

```diff
@@ -1,11 +1,21 @@
 import type { Replace } from '@/types/utils'
+import type { RouterLinkProps } from 'vue-router'
 
 export type ButtonAppearance = 'primary' | 'secondary' | 'tertiary' | 'danger' | 'none'
 export type ButtonAppearanceRecord = Record<Replace<ButtonAppearance, '-', ''>, ButtonAppearance>
 
 export type ButtonSize = 'large' | 'medium' | 'small'
 export type ButtonSizeRecord = Record<ButtonSize, ButtonSize>
 
+export type ButtonProps = {
```

Comment:
suggestion: if we can add TS block comments here for each prop, it would expose them should we ever be integrated with vue-component-meta

Reply:
Good idea!

Reply:
Added.